src/XrdHttp/XrdHttpReq.cc

```diff
@@ -2813,3 +2882,64 @@ void XrdHttpReq::getfhandle() {
           (int) fhandle[0] << ":" << (int) fhandle[1] << ":" << (int) fhandle[2] << ":" << (int) fhandle[3]);
 
 }
+
+std::string XrdHttpReq::EnumToString( int enumVal )
+{
+  switch(enumVal)
+  {
+    case XrdHttpReq::rtUnset:
+        return "Unset";
+    case XrdHttpReq::rtUnknown:
+        return "Unknown";
+    case XrdHttpReq::rtMalformed:
+        return "Malformed";
+    case XrdHttpReq::rtGET:
+        return "GET";
+    case XrdHttpReq::rtHEAD:
+        return "HEAD";
+    case XrdHttpReq::rtPUT:
+        return "PUT";
+    case XrdHttpReq::rtOPTIONS:
+        return "OPTIONS";
+    case XrdHttpReq::rtPATCH:
+        return "PATCH";
+    case XrdHttpReq::rtDELETE:
+        return "DELETE";
+    case XrdHttpReq::rtPROPFIND:
+        return "PROPFIND";
+    case XrdHttpReq::rtMKCOL:
+        return "MKCOL";
+    case XrdHttpReq::rtMOVE:
+        return "MOVE";
+    case XrdHttpReq::rtPOST:
+        return "POST";
+    default:
+        return "NOT_RECOGNIZED";
+  }
+}
+
+
+void XrdHttpReq::logTransferEvent(const int debug, const int retval, const char *message)
+{
+  if (retval != -1)
```

Comment:
Can we short circuit the creation of the `std::stringstream` if all debugging is disabled?.